components/SignUpForm.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { Field, reduxForm } from 'redux-form';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+
+const renderField = ({ input, label, type, meta: { touched, error } }) =>
+  <div>
+    <input {...input} placeholder={label} type={type} />
+    {touched &&
+      error &&
+      <div>
+        {error}
+      </div>}
+  </div>;
+
+const renderErrors = errors =>
+  <div className="alert alert-danger" role="alert">
+    {errors &&
+      errors.map(err =>
```

Comment:
idx not approved by eslint https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md